### ISSUE TITLE
Translate directive conflicts with html5 attribute

### ISSUE THREAD

**Issue by demirag** [angular-translate/angular-translate#849]
There is also html5 attribute 'translate=yes|no' to tell to translation services specific content should or should not be translated.  More info here http://www.w3.org/International/questions/qa-translate-flag

In current angular-translate version 2.5.0, attribute 'translate' is used for the directive. So it is impossible to use html5 attribute side by side. 

I think we need to attach a prefix to directive name or change it to something else.

**Comment by xsegrity** [comment]
I'm new to angularjs but would using the "data-" prefix not solve the issue (e.g. data-translate)?

**Comment by tspaeth** [comment]
as data- in angularjs is just an automate-scanned alternative I would bet that it still reacts on translate AND data-translate. So it might not really help with respect to "not using the translate attribute".

**Comment by xsegrity** [comment]
Ah right - that makes sense. Like I said < new guy :)

**Comment by creitzel** [comment]
+1 for wrapper in current major version.   It would help to give developers time to migrate their code in preparation for the next major version when support is dropped for the conflicting attribute name "translate".   

My suggestion for the new attribute name: "message-id".   The "-id" part reflects that the values are the ID of the actual text that is to be retrieved.

**Comment by knalli** [comment]
@PascalPrecht  @tspaeth Request for decision

I'm actually absolutely for migrating away from `translate` avoiding conflicts with existing HTML standard attributes. And it will be a no brainer adding an additional wrapper/alias for `translate` on demand (even for compat).

So, the actual question is: How to call it? Would be a `<h1 message="translationId"></h1>` and `{{ 'translationId` | message }}` ok? The term `messaging` using for translations is not new in other domains (i.e. Java/Spring and its `messageSource`), but it looks uncommon in web for me. Wdyt?

**Comment by tspaeth** [comment]
Hm, I wouldn't do it without a prefix as we can't be sure if not any other library or whatever server side framework adds additions (you might also find AJS in some ugly mixtures with JavaEE frameworks as we both don't want to know, but I've seen this somewhere... :-) ).
So I would advise more something like

```<h1 x-ngtrans-message="translationId"><h1>``` then or just ngtranslate or pptranslate ;-)

**Comment by knalli** [comment]
Speaking of extending, the prefixed `$vendor-` would be the best one? @PascalPrecht you are more familiar with webcomponents/polymer.

But it should not be too long after all. Must be still easy to use. I'd already the idea of `t7n`, but is not only short and unique but also very obfuscated.

**Comment by tspaeth** [comment]
you'll collide with t3n somehow (maybe only we germans know that term as a print publication).

**Comment by demirag** [comment]
I think safest option is to prefix this. "ng-translate" or better "pp-translate". For the backwards compatibility, we can still support "translate" attribute for a while to give time everyone to migrate.

**Comment by tspaeth** [comment]
even though you might not use these modules in parallel, it'll collide with https://www.npmjs.com/package/ng-translation if you use "ng-translate" :)

**Comment by demirag** [comment]
Yeah ng-translate might collide again any time. I would go for pp-translate to be more safe.

**Comment by knalli** [comment]
Conclusion of the last ones:

Standard

```html
<span pp-translate-cloak></span>
<span pp-translate-cloak="TEXT"></span>
<span pp-translate="TEXT"></span>
<span pp-translate>TEXT</span>
<span pp-translate-attr pp-translate-attr-title="TEXT"></span>
<span>{{ 'TEXT' | ppTranslate }}</span>
```

Compat

```html
<span translate-cloak></span>
<span translate-cloak="TEXT"></span>
<span translate="TEXT"></span>
<span translate>TEXT</span>
<span translate-attr translate-attr-title="TEXT"></span>
<span>{{ 'TEXT' | translate }}</span>
```

Indeed, would be streamlined with the package name.

**Comment by brownknows** [comment]
Just wondering if there is a release on the immediate horizon with this change.   Thanks.

**Comment by knalli** [comment]
I do not see this feature in the v2. And there is still no timeline (with time) for v3.

**Comment by knalli** [comment]
Note: Definitely not in v2 (implications are big). And there is still no plan for a v3 release.

Close this due #1671. Again: This is already part of the v3-agenda

**Comment by knalli** [comment]
Hello,

in part of a "Spring Cleaning 2017", we are closing this issue as part of several others. The meta discussion is available [here](https://github.com/angular-translate/angular-translate/issues/1671) for full transparency.

At the end, we are a small team and have only limited resources (both time and human) available. Said this, we must keep the focus and scope on the project.

We are grateful and appreciate to everyone for each made contribution (regardless being code, idea, documentation, and support).

We are deprecating/closing all issues which fulfills a specific set of criteria: mainly outdated, nice-to-have, or just out-of-scope.

----

Although the list has been curated to the best of our's knowledge, there will be some lack of understanding.

Okay, there a different stories behind all open issues, but we do not have the time for everything:

* Support: Because we have only limited resources, we cannot handle support questions any more. Please use appropriate boards like [Stackoverflow](http://stackoverflow.com/questions/tagged/angular-translate).
* Feature requests or ehancements: Altough it is probalbly a great idea (otherwise it would be closed already), we can not giving more resources to things out of scope or which are useful for too specific scenarios only.
* Bugs: Filed a long time ago, this bug either has a reasonable workaround, no effective solution without harming all others, or is lacking additional information/use-case.
* Docs: All documentation efforts not matching English will be abonded.

Thank you.

*angular-translate team*

----

# FAQ

**"But this is really important!"**

Well, we are all humans. Maybe we have overseen an important issue. If we missed something, we hope you are reaching as again *filing a new issue* with the reference to the old one. But please be aware: A bug without further details and a demo included will be rejected again.

**"But this would a great feature!"**

Reconsider: Are we really required to add something in order to make this feature happen? Or could you also create a project on your own utilizing this project?

**Comment by Sjors** [comment]
I found a workaround, though I haven't tested it extensively.

Usage, if you want the HTML 5 behavior:
```html
<span translate="no">Hello World</span>
```

Just make sure you don't use lower case `no` or `yes` for any real translation.

Custom directive:

```js
angular
  .module('yourApp')
  .directive('translate', translate);

function translate () {
  const directive = {
    restrict: 'A',
    compile: compile,
    priority: 1
  };
  return directive;

  function compile (elem, attrs) {
    if (attrs.translate === 'no' || attrs.translate === 'yes') {
      // HTML5 tag. Prevent angular-translate from processing this.
      attrs.translate = undefined;
    }
  }
}
```

The original directive should have a lower priority, e.g.:
```js
$translateProvider.directivePriority(0);
```
